core/base-service/base.js

```diff
@@ -106,7 +106,22 @@ module.exports = class BaseService {
    *  - capture: Array of names for the capture groups in the regular
    *             expression. The handler will be passed an object containing
    *             the matches.
-   *  - queryParamSchema: Joi schema for valid query params.
+   *  - queryParamSchema: (Optional) A Joi schema (`Joi.object({ ... }).required()`)
+   *                      for the query param object. If you know a parameter
+   *                      will never receive a numeric string, you can use
+   *                      `Joi.string()`. Because of quirks in Scoutcamp and Joi,
+   *                      alphanumeric strings should be declared using
+   *                      `Joi.alternatives().try(Joi.string(), Joi.number())`,
+   *                      otherwise a value like `?success_color=999` will fail.
```

Comment:
To clarify, we need to use `Joi.alternatives().try(Joi.string(), Joi.number())` because `?success_color=999` would fail on `Joi.number()` ?

Reply:
Never mind, based on the below content in the docs I believe it's the latter case 👍